doc/api/cli.md

```diff
@@ -2588,6 +2588,44 @@ added: v0.8.0
 
 Print stack traces for deprecations.
 
+### `--trace-env`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+Print reads and writes to environment variables.
+
+The following accesses will be printed:
+
+* The environment variable reads that Node.js does internally.
+* Writes in the form of `process.env.KEY = "SOME VALUE"`.
+* Reads in the form of `process.env.KEY`.
+* Definitions in the form of `Object.defineProperty(process.env, 'KEY', {...})`.
+* Queries in the form of `Object.hasOwn(process.env, 'KEY')`,
+  `process.env.hasOwnProperty('KEY')` or `'KEY' in process.env`.
+* Deletions in the form of `delete process.env.KEY`.
+* Enumerations inf the form of `...process.env` or `Object.keys(process.env)`.
```

Comment:
This will potentially unvail credentials in the env (e.g. dotenv) once it has been accessed to the stdout. Would this be a security issue that is worth mentioning?

Reply:
hmm, yeah this is potentially a bit problematic if folks don't expect it. Perhaps by default the `--trace-env*` flags shouldn't actually include the values (or should replace the values with something like `****`, and have the user explicitly use an option like `--trace-env=values` to explicitly opt-in to printing the values for each key.

Reply:
Updated to just not print the values at all - at least for the use cases intended so far (checking consistency), this does not seem to be particularly important. We can add another flag or something to print the value when someone requests it.